request/builder/base.go

```diff
@@ -225,13 +226,25 @@ func (b *BaseBuilder) parseRequestURL() error {
 
 func (b *BaseBuilder) setupHeaders(httpRequest *http.Request) error {
 	if b.parsedHeaders != nil {
+
 		for headerKey, headerValue := range *b.parsedHeaders {
-			for _, r := range headerValue {
-				if r > unicode.MaxASCII {
-					headerValue = utils.URLQueryEscape(headerValue)
-					break
+			if headerKey == "X-QS-Fetch-Source" {
+				// header X-QS-Fetch-Source is a URL to fetch.
+				// We should first parse this URL.
+				requestURL, err := url.Parse(headerValue)
+				if err != nil {
+					return errors.New("Invalid http header value")
```

Comment:
It's better to return the detail.

`return fmt.Errorf("invalid HTTP header value: %s", headerValue)`